servers/store/src/main/java/org/projectnessie/server/store/TableCommitMetaStoreWorker.java

```diff
@@ -45,7 +47,142 @@
+      builder.setIcebergTableMetadata(stateBuilder);
+    } else {
+      throw new IllegalArgumentException("Unknown type " + contents);
+    }
+
+    return builder.build().toByteString();
+  }
+
+  @Override
+  public Contents valueFromStore(ByteString onReferenceValue, Optional<ByteString> globalState) {
+    ObjectTypes.Contents contents = parse(onReferenceValue);
+    Optional<ObjectTypes.Contents> globalContents =
+        globalState.map(TableCommitMetaStoreWorker::parse);
+    switch (contents.getObjectTypeCase()) {
+      case DELTA_LAKE_TABLE:
+        Builder builder =
+            ImmutableDeltaLakeTable.builder()
+                .id(contents.getId())
+                .addAllMetadataLocationHistory(
+                    contents.getDeltaLakeTable().getMetadataLocationHistoryList())
+                .addAllCheckpointLocationHistory(
+                    contents.getDeltaLakeTable().getCheckpointLocationHistoryList());
+        if (contents.getDeltaLakeTable().getLastCheckpoint() != null) {
+          builder.lastCheckpoint(contents.getDeltaLakeTable().getLastCheckpoint());
+        }
+        return builder.build();
+
+      case ICEBERG_SNAPSHOT:
+        return IcebergTable.of(
+            globalContents
+                .map(ObjectTypes.Contents::getIcebergTableMetadata)
+                .map(IcebergTableMetadata::getMetadataLocation)
+                .orElseThrow(IllegalStateException::new),
+            contents.getIcebergSnapshot().getSnapshotId(),
+            contents.getId());
+
+      case SQL_VIEW:
+        ObjectTypes.SqlView view = contents.getSqlView();
+        return ImmutableSqlView.builder()
+            .dialect(Dialect.valueOf(view.getDialect()))
+            .sqlText(view.getSqlText())
+            .id(contents.getId())
+            .build();
+
+      case OBJECTTYPE_NOT_SET:
+      default:
+        throw new IllegalArgumentException("Unknown type " + contents.getObjectTypeCase());
+    }
+  }
+
+  @Override
+  public String getId(Contents contents) {
+    return contents.getId();
+  }
+
+  @Override
+  public Byte getPayload(Contents contents) {
+    if (contents instanceof IcebergTable) {
+      return (byte) Contents.Type.ICEBERG_TABLE.ordinal();
+    } else if (contents instanceof DeltaLakeTable) {
+      return (byte) Contents.Type.DELTA_LAKE_TABLE.ordinal();
+    } else if (contents instanceof SqlView) {
+      return (byte) Contents.Type.VIEW.ordinal();
+    } else {
+      throw new IllegalArgumentException("Unknown type " + contents);
+    }
+  }
+
+  @Override
+  public Type getType(Byte payload) {
+    if (payload == null || payload > Contents.Type.values().length || payload < 0) {
+      throw new IllegalArgumentException(
+          String.format("Cannot create type from payload. Payload %d does not exist", payload));
+    }
+    return Contents.Type.values()[payload];
+  }
+
+  @Override
+  public boolean requiresState(Contents contents) {
+    return contents instanceof IcebergTable;
```

Comment:
nit: `requiresGlobalState`, perhaps, for the same of clarity?

Reply:
makes sense, yes